### PR TITLE
Code redesign readme

### PR DESCRIPTION
**Pull request by oferShinaar**
adding support for for demos and nexys-a7

### CODE REVIEW

**Review by ronen-haen-sndk — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -85,9 +91,17 @@ WD-Firmware
 	- In the Import window select *General* -> *Existing Projects into Workspace* -> *Next*
 	- In the next Import window *Select root directory* -> *Browse*  and choose the infra-riscv-fw/WD-Firmware/ you've downloaded in 'Getting the firmware sources' section
 	- Press *'Finish'* button
-- From 'Eclipse MCU' menu bar select '*Project'* -> *'Build All'*
+- From 'Eclipse MCU' menu bar select '*Project'* -> *'Build All'*. Note that you can select which platform to build for.
+- You you will need to choose a specific ***demo*** for building a full sultion:
+
+        from the eclipse terminal or console:
+        $ cd [WD-firmware-root]/WD-Firmware/demo/build
+        $ python ./config.sh
```
Comment (nati-rapaport) [inline README.md]: You need to run ./config.sh, not as written here - python ./config.sh

File: README.md [hunk 2]
```diff
@@ -35,24 +35,26 @@ The structure of WD Firmware package allows quick and easy integration for more
...
-          |-rtosal                          <-- rtos abstraction layer
-          |-rtos_core                       <-- specific rtos source code
-                |-freertos
-                |-....
+     ├── board                                <-- supported boards
+          ├── hifive-1                        
+          ├── ihfive-unleashed (not supported yet)
+          ├── nexys_a7 (Support for SweRV v1)
+     ├── common                               <-- common source
+     ├── demos                                <-- demos source
```
Comment (nati-rapaport) [inline README.md]: "demo", not "demos"

File: README.md [hunk 3]
```diff
@@ -35,24 +35,26 @@ The structure of WD Firmware package allows quick and easy integration for more
...
-                |-....
+     ├── board                                <-- supported boards
+          ├── hifive-1                        
+          ├── ihfive-unleashed (not supported yet)
+          ├── nexys_a7 (Support for SweRV v1)
+     ├── common                               <-- common source
+     ├── demos                                <-- demos source 
+          ├── build                           <-- examples build scripts
+          ├── demo_rtosal.c                   <-- Abstruction Layer (AL) demo on FreeRTOS
+          ├── main.c                          <-- The main of all demos
```
Comment (nati-rapaport) [inline README.md:61]: there are also the "demo_platform_al.h+c files, don't you want to mention them here too?

File: README.md (hunk 1 above)
Comment (oferShinaar) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -85,9 +91,17 @@ WD-Firmware
...
-- From 'Eclipse MCU' menu bar select '*Project'* -> *'Build All'*
+- From 'Eclipse MCU' menu bar select '*Project'* -> *'Build All'*. Note that you can select which platform to build for.
+- You you will need to choose a specific ***demo*** for building a full sultion:
+
+        from the eclipse terminal or console:
+        $ cd [WD-firmware-root]/WD-Firmware/demo/build
+        $ python ./config.sh
+        
+        Then you will be asked to choose a demo.
+    For more explnation on adding new demos please read the readme file on ***'/demo'***
```
Comment (nati-rapaport) [inline README.md]: In order to build there is also a need to follow one of the alternatives:
First option (using eclipse)
1) right click on WD-Firmware project caption
2) From the popping list, select: Build Configuration -> Set Active -> hifive1
3) Click Project -> Build all (or just Ctrl+B)

Second option (from console)
1) $ cd /WD-Firmware/demo/build
2) $ scons target=hifive1 + enter

Actually you mentioned the build before, but the correct order is to first select the demo and then build

File: README.md (hunk 3 above)
Comment (oferShinaar) [reply README.md:61]: nope. no need it is just tree layout

File: README.md [hunk 5]
```diff
@@ -105,32 +119,52 @@ WD-Firmware
...
+
+            - hifive1
+            - nexys_a7_Swerv1
+            
+
 
 ### Adding new source modules
 
-The folder WD-Firmware/examples/build/ contains a template file (SConscript_template) which can be used.
+The folder WD-Firmware/demos/build/ contains a template file (SConscript_template) which can be used.
```
Comment (nati-rapaport) [inline README.md]: WD-Firmware/demo/build/  not 'demos'

File: README.md [hunk 6]
```diff
@@ -71,6 +73,10 @@ WD-Firmware
       $ sudo ln -s /usr/lib/x86_64-linux-gnu/libmpfr.so.6 /usr/lib/x86_64-linux-gnu/libmpfr.so.4 
 - Download and install Java SE Runtime Environment
 
+- For RISC-V OpenOCD, you will need to following depended libs: libusb-0.1, libusb-1.0-0-dev, libusb-dev
```
Comment (nati-rapaport) [inline README.md]: "you will need **the** following", not "you will need **to** following"

File: README.md (hunk 4 above)
Comment (oferShinaar) [reply README.md]: no need, this is Eclipse knowledge. we provide simple build.

File: README.md (hunk 2 above)
Comment (oferShinaar) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (oferShinaar) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (oferShinaar) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (oferShinaar) [reply README.md]: switched between bullets

File: README.md [hunk 7]
```diff
@@ -28,31 +28,44 @@ This repo is WD RISC-V Firmware package, holds:
...
-          |-rtos_core                       <-- specific rtos source code
-                |-freertos
-                |-....
+     ├── board                                <-- supported boards
+          ├── hifive-1                        
+          ├── ihfive-unleashed (not supported yet)
+          ├── nexys_a7 (Support for SweRV v1)
+     ├── common                               <-- common source
+     ├── demo                                 <-- demos source 
+          ├── build                           <-- examples build scripts
```
Comment (nati-rapaport) [inline README.md]: <-- example build scripts rather than <-- examples build scripts

**Review by nati-rapaport — APPROVED** [review]

File: README.md (hunk 7 above)
Comment (oferShinaar) [reply README.md]: done 92f57d6

**Pull request merged**